### PR TITLE
Add initial draft of Netnology's SDWAN Datadog integration

### PR DESCRIPTION
**Pull request by nikhilguptanetnologyio**
### What does this PR do?

Adds the necessary files to add the nn_sdwan integration to the repo.

### Motivation

Need some guidance on resolving a development/testing issue that we are running into (described below).

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

We think there is some kind of packaging/namespacing difference between the datadog-agent and the ddev environment that we don't know how to bridge.

Our main file, `check.py`, imports our other source code file `net_api.py`. If it imports it via `from net_api import vManageApi`, then we copy these files over to the datadog-agent at `/etc/datadog-agent/checks.d/`, then the integration works fine but ddev tests don't work:

``` bash
$ ddev test nn_sdwan
...
E   ModuleNotFoundError: No module named 'net_api'
...
ERROR:   py39: commands failed
```

If instead we import it via `from .net_api import vManageApi` and copy over the files the same way, then the integration doesn't work with datadog-agent anymore:

``` bash
$ datadog-agent check nn_sdwan
...
error: Unable to load a check from instance of config 'nn_sdwan'
...
 unable to import module 'nn_sdwan': Traceback (most recent call last):
  File "/etc/datadog-agent/checks.d/nn_sdwan.py", line 8, in <module>
    from .net_api import vManageApi
...
ImportError: attempted relative import with no known parent package

Error: no valid check found
```

...but now it does work just fine with tests.

Regardless of what we do, the command `ddev env check nn_sdwan py39` produces the same error every time `Error: no valid check found` with no suggestions on how to fix that, which is another issue throwing us off.

### CODE REVIEW

File: nn_sdwan/tox.ini
Comment (hithwen) [inline nn_sdwan/tox.ini:13]:
```suggestion
    py38: {toxworkdir}/py38
```
It seems you're using py3 syntax so the tests fail to run with python2.
Also the agent does not still support py3.9

File: nn_sdwan/setup.py
Comment (hithwen) [inline nn_sdwan/setup.py:54]:
```suggestion
```
It seems this integration is written using py3 syntax

**Comment by hithwen** [comment]
Hallo, aside from the changes mentioned you also need to run the following command and commit the changes generated.
`ddev validate ci --fix`

**Comment by nikhilguptanetnologyio** [comment]
Thanks, I've implemented those changes you suggested. They did not resolve the issues described earlier though.

**Comment by yzhan289** [comment]
Hey @nikhilguptanetnologyio 👋 , `from .net_api import vManageApi` is the correct way to import other source code. 

To test out your unreleased check in a real agent, the steps are the following:
1. Run `ddev -e release build nn_sdwan` to build a python wheel of the check manually.
2. Run `datadog-agent integration install -w <PATH_OF_NN_SDWAN_ARTIFACT>/<NN_SDWAN_ARTIFACT_NAME>.whl` to install the wheel onto your datadog agent.

When you received `Error: no valid check found` after running `ddev env check nn_sdwan py39`, make sure to run `ddev env start nn_sdwan py39` first: https://datadoghq.dev/integrations-core/ddev/cli/#ddev-env-start.

File: nn_sdwan/README.md
Comment (anawishnoff) [inline nn_sdwan/README.md:5]:
```suggestion
This check monitors SDWAN controllers through the Datadog Agent using a SDWAN platorm provided by [Netnology][1]. Currently only Cisco vManage devices are supported.
```

This might be a bit clearer?

Comment (anawishnoff) [inline nn_sdwan/README.md:5]:
```suggestion
This check monitors SDWAN controllers through the Datadog Agent using a SDWAN platorm provided by [Netnology][1]. Currently only Cisco vManage devices are supported.
```

This might be a bit clearer?

File: nn_sdwan/README.md
Comment (anawishnoff) [inline nn_sdwan/README.md:9]: I would spell out the full name of the integration here

File: nn_sdwan/README.md
Comment (anawishnoff) [inline nn_sdwan/README.md:6]: It would be helpful to add a bit more context here about what kinds of data is collected by this integration.

File: nn_sdwan/manifest.json
Comment (anawishnoff) [inline nn_sdwan/manifest.json:18]: If you want to add categories to your listing, you can find the list at the top of this page: [https://docs.datadoghq.com/integrations/](https://docs.datadoghq.com/integrations/)

File: nn_sdwan/assets/dashboards/nn_sdwan_overview.json
Comment (anawishnoff) [inline nn_sdwan/assets/dashboards/nn_sdwan_overview.json:1]: For partner-built integrations, we usually recommend adding a logo and some descriptive text to the dashboard so customers can easily understand what they're being shown.

**Review by anawishnoff — CHANGES_REQUESTED** [review]
Hi @nikhilguptanetnologyio, I left a few comments after reviewing your PR. Let me know if you have any questions!

**Comment by yzhan289** [comment]
@nikhilguptanetnologyio Apologies, when testing a new integration, make sure to also have `--dev` in `ddev env start nn_sdwan py39 --dev` since it will run the check in your repo rather than the one shipped with the agent. 

Also, when you test with `ddev`, make sure to also add py27 and py39 to the `testenv` in the `tox.ini` file in order to get them working:

```
[testenv]
ensure_default_envdir = true
envdir =
    py27: {toxworkdir}/py27
    py38: {toxworkdir}/py38
    py39: {toxworkdir}/py39
```

**Comment by nikhilguptanetnologyio** [comment]
This is what happens when I run it with `--dev` at the end @yzhan289 :

```
[integrations-extras]# ddev env start nn_sdwan py38 --dev
Environment variable DD_API_KEY does not exist; a well-formatted fake API key will be used instead. You can also set the API key by doing `ddev config set dd_api_key`.
Setting up environment `py38`... success!
Updating `datadog/agent-dev:master-py3`... success!
Detecting the major version... Agent 7 detected
Writing configuration for `py38`... success!
Starting the Agent... success!
Upgrading `nn_sdwan` check to the development version... Traceback (most recent call last):
  File "/root/dd/integrations-extras/venv/bin/ddev", line 8, in <module>
    sys.exit(ddev())
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/datadog_checks/dev/tooling/commands/env/start.py", line 79, in start
    dev_or_base = _perform_updates(base, dev, environment, check)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/datadog_checks/dev/tooling/commands/env/start.py", line 290, in _perform_updates
    environment.update_check()
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/datadog_checks/dev/tooling/e2e/docker.py", line 274, in update_check
    run_command(command, capture=True, check=True)
  File "/root/dd/integrations-extras/venv/lib64/python3.8/site-packages/datadog_checks/dev/subprocess.py", line 77, in run_command
    raise SubprocessError(
datadog_checks.dev.errors.SubprocessError: Command: ['docker', 'exec', 'dd_nn_sdwan_py38', '/opt/datadog-agent/embedded/bin/pip3', 'install', '-e', '/home/nn_sdwan[deps]']
Exit code: 1
Captured Output: Obtaining file:///home/nn_sdwan
ERROR: file:///home/nn_sdwan does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

Looks like the container it created is missing a file/folder... But it definitely exists outside the container because it's a file in this PR and immediately before/after I can see our `setup.py` file is there.
``` bash
[integrations-extras]# ls nn_sdwan/setup.py
nn_sdwan/setup.py
```

File: nn_sdwan/metadata.csv
Comment (urseberry) [inline nn_sdwan/metadata.csv:1]: Please add descriptions to the fields below.

**Comment by nikhilguptanetnologyio** [comment]
Thanks @anawishnoff and @urseberry, I think I've addressed all the feedback.

@yzhan289 Do have any other suggestions to resolve the `ddev env start nn_sdwan py38 --dev` issue? The integration seems to be working fine when building/installing the wheel, so is this an issue we should worry about?

File: nn_sdwan/manifest.json
Comment (steveny91) [inline nn_sdwan/manifest.json:30]: This should either be:
```suggestion
      "Netnology SDWAN Overview": "assets/dashboards/nn_sdwan_overview.json"
```
or you'd have to rename the json in assets/dashboard to `overview.json`

File: nn_sdwan/manifest.json
Comment (steveny91) [inline nn_sdwan/manifest.json:11]:
```suggestion
  "support": "contrib",
```
`extras` is not valid here. The choices are 'core', 'contrib', 'partner'

**Review by steveny91 — CHANGES_REQUESTED** [review]
Hello! 

Would you mind running the following:
`ddev -e validate config nn_sdwan -s` 
`ddev -e validate models nn_sdwan -s`

This will resync the config file.

Aside from that, I see that you're also missing the field `metric_to_check` in the manifest.json. This should be a list of metric that we can use to determine if the integration is working properly or not. This has to be a metric that is listed in the metadata.csv. [Example](https://github.com/DataDog/integrations-core/blob/master/activemq/manifest.json#L15-L18).

In the two json files containing the recommended monitors definitions, there's a required param for `recommended_monitor_metadata`. [Example](https://github.com/DataDog/integrations-core/blob/master/activemq/assets/recommended_monitors/activemq_artemis_high_disk_store.json#L27-L29)

File: nn_sdwan/assets/monitors/high-latency-monitor.json
Comment (steveny91) [inline nn_sdwan/assets/monitors/high-latency-monitor.json:2]: Not sure what the `id` is, this should be deleted

File: nn_sdwan/assets/monitors/high-latency-monitor.json
Comment (steveny91) [inline nn_sdwan/assets/monitors/high-latency-monitor.json:7]: Should have a tag in the form of `integration:<integration_name>`. Example:
```suggestion
    "tags": [
      "integration:nn_sdwan"
    ],
```

File: nn_sdwan/assets/monitors/high-latency-monitor.json
Comment (steveny91) [inline nn_sdwan/assets/monitors/high-latency-monitor.json:3]: Should be name of integration 
```suggestion
    "name": "nn_sdwan",
```

File: nn_sdwan/assets/monitors/packet-loss-monitor.json
Comment (steveny91) [inline nn_sdwan/assets/monitors/packet-loss-monitor.json:2]: same as above

File: nn_sdwan/assets/monitors/packet-loss-monitor.json
Comment (steveny91) [inline nn_sdwan/assets/monitors/packet-loss-monitor.json:3]: same as above

File: nn_sdwan/assets/monitors/packet-loss-monitor.json
Comment (steveny91) [inline nn_sdwan/assets/monitors/packet-loss-monitor.json:7]: same as above

File: nn_sdwan/assets/service_checks.json
Comment (steveny91) [inline nn_sdwan/assets/service_checks.json:4]: must match manifest display_name `Netnology SDWAN` 
```suggestion
        "integration": "Netnology SDWAN,
```

File: nn_sdwan/assets/service_checks.json
Comment (steveny91) [inline nn_sdwan/assets/service_checks.json:12]: Must match manifest display_name `Netnology SDWAN` 
```suggestion
        "name": "Netnology SDWAN",
```

File: nn_sdwan/assets/monitors/high-latency-monitor.json
Comment (nikhilguptanetnologyio) [reply nn_sdwan/assets/monitors/high-latency-monitor.json:2]: Oh I think this is because I created the monitor under my account, then exported a fully created/registered monitor. So there are fields related to my account/org/permissions/etc. Exporting the monitor before officially creating it in my account seems to fix this.

File: nn_sdwan/assets/monitors/high-latency-monitor.json
Comment (nikhilguptanetnologyio) [reply nn_sdwan/assets/monitors/high-latency-monitor.json:3]: If I change this then the monitor itself would be named "nn_sdwan" when a customer/user is browsing their monitor list. They wouldn't know what it actually does or alerts on, unless they go inside and inspect the configuration.

Is that not what would happen here? If so, seems counterintuitive.

**Review by urseberry — APPROVED** [review]

**Comment by nikhilguptanetnologyio** [comment]
I've integrated all the suggestions that @steveny91 had, except for the ones recommending the `name` of the service check and monitors to be changed. I think renaming them as suggested would make it more difficult for customers to use the integration, because then it would not be clear what each monitor or the service check actually does. The [`activemq` example (as linked earlier)](https://github.com/DataDog/integrations-core/blob/master/activemq/assets/recommended_monitors/activemq_artemis_high_disk_store.json) doesn't follow that guideline, so I'm not sure whether to make the change or not.

**Comment by nikhilguptanetnologyio** [comment]
I went ahead and made the naming changes as well.

File: nn_sdwan/README.md
Comment (anawishnoff) [inline nn_sdwan/README.md:1]: This should match the integration name:
```suggestion
# Agent Check: Netnology SDWAN
```

**Review by anawishnoff — APPROVED** [review]

File: nn_sdwan/tox.ini
Comment (fanny-jiang) [inline nn_sdwan/tox.ini:6]:
```suggestion
    py{38,39}
```
Do we need to test on py27? The py27 tests failed due to presence of python 3 syntax. Also, should the base python be py38 or are you leaving it as py39 for when the agent bumps the python version?

**Review by fanny-jiang — CHANGES_REQUESTED** [review]
Added some comments. Also, are you still having trouble running/starting the e2e agent? Have you configured `ddev` to use the integrations-extras repo and use the `-e/--extras` flag? 

```
ddev config set repos.extras /path/to/integrations-extras
ddev config set repo extras
```

`ddev -e env start nn_sdwan py38 --dev`

File: .azure-pipelines/all.yml
Comment (nikhilguptanetnologyio) [reply .azure-pipelines/all.yml:35]: These weren't changed deliberately. According to the commit history, this happened on bc05e8863, which is when I ran `ddev validate ci --fix` for the first time.

File: nn_sdwan/tox.ini
Comment (nikhilguptanetnologyio) [reply nn_sdwan/tox.ini:6]: No, I don't think we need to test on py27.

Originally we were developing this in a py39 environment, but then were told that the agent doesn't support py39 so now we're using a py38 environment. I believe the integration works the same in both, though. What do you suggest we change `basepython` and/or `envlist` to?

Comment (fanny-jiang) [reply nn_sdwan/tox.ini:6]: Got it. Let's use `basepython = py38` and `envlist` py38 

```suggestion
    py38
```

File: .azure-pipelines/all.yml
Comment (fanny-jiang) [reply .azure-pipelines/all.yml:35]: Ah thanks for clarifying. I think the changes should be ok

File: nn_sdwan/datadog_checks/nn_sdwan/check.py
Comment (fanny-jiang) [inline nn_sdwan/datadog_checks/nn_sdwan/check.py:50]: Style tests are failing here: `./datadog_checks/nn_sdwan/check.py:50:121: E501 line too long (122 > 120 characters)`
Can you try running `ddev -e test nn_sdwan -fs`?

**Comment by fanny-jiang** [comment]
The conf.yaml.example also needs to be synced: `ddev -e validate config nn_sdwan -s`
And let's re-sync the config models too, just in case `ddev -e validate models nn_sdwan -s`

**Comment by anawishnoff** [comment]
Hi @nikhilguptanetnologyio, can you please add the Netnology logo to this PR as well so we can get it attached to your tile? Thanks!

**Comment by nikhilguptanetnologyio** [comment]
Thanks @fanny-jiang - all changes have been incorporated. Unfortunately, the `ddev -e env start nn_sdwan py38 --dev` command still produces the same error as [described before](https://github.com/DataDog/integrations-extras/pull/1244/#issuecomment-1086289718), even after setting the extras repo.

**Comment by nikhilguptanetnologyio** [comment]
Hi @anawishnoff I've added our logo as a .png file at `nn_sdwan/images/netnology-logo.png` .

**Comment by anawishnoff** [comment]
Thanks @nikhilguptanetnologyio! I'll send in a request to get this attached to your tile once it goes live.

File: .azure-pipelines/all.yml
Comment (sarah-witt) [reply .azure-pipelines/all.yml:35]: If you merge your branch with the latest changes from master, these changes should go away.

File: nn_sdwan/tests/test_nn_sdwan.py
Comment (sarah-witt) [inline nn_sdwan/tests/test_nn_sdwan.py:57]: This test is failing in the CI:
```

E     Asserted Metrics:
E     	- nn_sdwan.app_aware_routing.jitter
E     	- nn_sdwan.app_aware_routing.latency
E     	- nn_sdwan.app_aware_routing.loss_percentage
E     	- nn_sdwan.app_aware_routing.rx_octets
E     	- nn_sdwan.app_aware_routing.tx_octets
E     	- nn_sdwan.cert_summary
E     	- nn_sdwan.connection_summary_stats_error
E     	- nn_sdwan.connection_summary_stats_total
E     	- nn_sdwan.device_control_status
E     	- nn_sdwan.reboot_count
E     	- nn_sdwan.site_health
E     	- nn_sdwan.top_app_stats
E     	- nn_sdwan.transport_interface
E     	- nn_sdwan.vmanage_count
E     	- nn_sdwan.wan_edge_health
E     	- nn_sdwan.wan_edge_inventory
E     Found metrics that are not asserted:
E     	- 
```

These metrics do not appear to be emitted in the test, so the test is failing since they are asserted but not emitted in `assert_all_metrics_covered`. Typically a docker environment or test fixtures will need to be added to properly collect the metrics in tests. You can either look into this, or stop asserting metrics (which will decrease code coverage).

**Review by sarah-witt — COMMENTED** [review]
Hi @nikhilguptanetnologyio, I left a few comments. Thanks!

File: nn_sdwan/tests/test_nn_sdwan.py
Comment (nikhilguptanetnologyio) [reply nn_sdwan/tests/test_nn_sdwan.py:57]: Unfortunately, we can't put our required environment (SD-WAN controllers) in a docker environment, so I suppose we'll stop asserting the metrics.

File: nn_sdwan/datadog_checks/nn_sdwan/check.py
Comment (steveny91) [inline nn_sdwan/datadog_checks/nn_sdwan/check.py:56]: Would it make sense to try wrapping this in a try for error handling? Incase any of the function fails, then at least we don't throw an exception.

**Comment by steveny91** [comment]
Hello! Thanks for making all those changes. Glad everything is passing CI now. The only other few changes I would like to recommend, is to change the way we access the dictionaries. 

For example this [function](https://github.com/DataDog/integrations-extras/blob/5c23e783d9f74d90616f98185a37588bfde84d6c/nn_sdwan/datadog_checks/nn_sdwan/check.py#L68-L71). Using the following line as an example:

`wan_edge_inven_stats = self.vmanage_api.get_wan_edge_inventory()['data']`

The above line looks pretty brittle as the [function it calls](https://github.com/DataDog/integrations-extras/blob/5c23e783d9f74d90616f98185a37588bfde84d6c/nn_sdwan/datadog_checks/nn_sdwan/net_api.py#L123-L130), could potentially return an empty dictionary(`{}`) if the response code is not 200.  I would instead of accessing the dictionary with `dict[key]` try maybe using `dict.get(key)`.

This of course is coming from very little knowledge of the API we're communicating with.

File: nn_sdwan/datadog_checks/nn_sdwan/net_api.py
Comment (ofek) [inline nn_sdwan/datadog_checks/nn_sdwan/net_api.py:17]: Clever 😄

**Review by ofek — CHANGES_REQUESTED** [review]

**Comment by anawishnoff** [comment]
@nikhilguptanetnologyio thank you for making these changes! @ofek is there anything else needed from the agent integrations side?

**Comment by yzhan289** [comment]
Hey @nikhilguptanetnologyio , were you able to start a `ddev` environment in the end? I was able to start an environment on my local machine:

```
andrew.zhang@COMP-C02F513WML87 integrations-extras % ddev env start nn_sdwan py38 --dev
Setting up environment `py38`... success!
Updating `datadog/agent-dev:master-py3`... success!
Detecting the major version... Agent 7 detected
Writing configuration for `py38`... success!
Starting the Agent... success!
Upgrading `nn_sdwan` check to the development version... success!
Reloading the environment to reflect changes... success!

To edit config file, do: ddev env edit nn_sdwan py38
Config file (copied to your clipboard): /Users/andrew.zhang/Library/Application Support/dd-checks-dev/envs/nn_sdwan/py38/config/nn_sdwan.yaml
To run this check, do: ddev env check nn_sdwan py38
To stop this check, do: ddev env stop nn_sdwan py38
```

Running the check resulted in: 

```
  Running Checks
  ==============
    
    nn_sdwan (1.0.0)
    ----------------
      Instance ID: nn_sdwan:5087424ebc493eb6 [ERROR]
      Configuration Source: file:/etc/datadog-agent/conf.d/nn_sdwan.d/nn_sdwan.yaml
      Total Runs: 1
      Metric Samples: Last Run: 0, Total: 0
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 47ms
      Last Execution Date : 2022-05-31 14:54:12 UTC (1654008852000)
      Last Successful Execution Date : Never
      Error: get_subprocess_output expected output but had none.
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/base.py", line 1120, in run
          self.check(instance)
        File "/home/nn_sdwan/datadog_checks/nn_sdwan/check.py", line 30, in check
          if not self.pingable(self.instance.get('hostname')):
        File "/home/nn_sdwan/datadog_checks/nn_sdwan/check.py", line 67, in pingable
          out, err, retcode = get_subprocess_output(command, self.log)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/subprocess_output.py", line 56, in get_subprocess_output
          out, err, returncode = subprocess_output(cmd_args, raise_on_empty_output, env=env)
      _util.SubprocessOutputEmptyError: get_subprocess_output expected output but had none.
```
Looks like the `get_subprocess_output()` can raise a `SubprocessOutputEmptyError` on empty output: https://github.com/DataDog/integrations-core/blob/cfd04ffcaed83b58f501d54e575371228ed727ed/datadog_checks_base/datadog_checks/base/utils/subprocess_output.py#L33-L34. 

Also, have you rebased from `master` yet? I think rebasing should eliminate some of the unnecessary changes in `all.yml`.

**Comment by conikolg** [comment]
@yzhan289 We're already on the latest version (16.2.1) - I remember needing to update it recently because the github checks were producing different results than our locally installed `ddev`. So the issue has persisted through the last 2-3 months of updates.

**Comment by nikhilguptanetnologyio** [comment]
@yzhan289 We're already on the latest version (16.2.1) - I remember needing to update it recently because the github checks were producing different results than our locally installed `ddev`. So the issue has persisted through the last 2-3 months of updates.

**Review by yzhan289 — APPROVED** [review]
The ddev issue is likely just an issue with the repo/setup. Since I was able to start an environment on my own setup, I don't think it should be blocking the merge. LGTM otherwise!

File: nn_sdwan/MANIFEST.in
Comment (nmuesch) [inline nn_sdwan/MANIFEST.in:1]: @yzhan289 It looks like this one may have been made before the full pyproject migration, would you be able to help bring this up to date? My understanding is that will be required before this can be merged.

File: nn_sdwan/manifest.json
Comment (nmuesch) [inline nn_sdwan/manifest.json:26]: Is this intentional, it will need to be `true` when merged to be able to see it within Datadog documentation.

**Review by nmuesch — CHANGES_REQUESTED** [review]
Left a few minor comments but otherwise looks good. 

Can you also remove the empty `nn_sdwan/images/.gitkeep` file?

File: nn_sdwan/MANIFEST.in
Comment (yzhan289) [reply nn_sdwan/MANIFEST.in:1]: Communicated offline, but we can get this PR merged in now, and then @nikhilguptanetnologyio can submit a new PR with the pyproject migration.

**Comment by yzhan289** [comment]
> Hi @nikhilguptanetnologyio, one more comment - it looks like the title of the integration is getting cut off on the tile in our testing environment: Screen Shot 2022-06-02 at 11 12 59 AM
> 
> Would you be able to shorten the title (perhaps just removing the word "integration")?

I think this is automatically added to all integrations. Is it possible to remove any of the other words?

File: nn_sdwan/manifest.json
Comment (nikhilguptanetnologyio) [reply nn_sdwan/manifest.json]: Not intentional, maybe that's the default value when the project is created by `ddev`? Either way, fixed.
I also removed `nn_sdwan/images/.gitkeep`.

**Comment by nikhilguptanetnologyio** [comment]
@anawishnoff @yzhan289 Based on `manifest.json`, our `display_name` is `Netnology SD-WAN`. Our `public_title` is `Datadog-Netnology SD-WAN Integration`. Looks like the display name is the problem - it's too long? Unfortunately I don't think we can make that shorter; one word is our company name and the other is the (broad) use-case that the integration is made for.

I'm looking at the public integration page right now, and many have "Integration" at the end, but not all (like `Dell Inc.`, `zabbix`, `Docker`, etc). Some of those that do have the word "Integration" are overflowing on to the next line, but "Integration" is hidden by the "Available" button, like `Amazon ECS on AWS Fargate Integration` or `Windows Kernel Memory Integration`. There's an interesting one I've noticed called `Windows performance counters` that overflows yet doesn't have "Integration" at the end.

File: nn_sdwan/manifest.json
Comment (yzhan289) [inline nn_sdwan/manifest.json:20]:
```suggestion
  "public_title": "Datadog-Netnology SD-WAN",
```
Non-blocking comment, if you did want to the shorten the name. It looks like you can override this value to fix the name, similar to windows performance counters: https://github.com/DataDog/integrations-core/blob/master/windows_performance_counters/manifest.json#L15

**Review by yzhan289 — APPROVED** [review]

**Review by yzhan289 — APPROVED** [review]

**Pull request merged**
